frontend/src/widgets/RegionalGoalDashboard/TopicsTable.js

```diff
@@ -0,0 +1,181 @@
+ */
+function TopicRow({ topic, statuses, total }) {
+  return (
+    <tr key={topic}>
+      <td>{topic}</td>
+      <td>{statuses['Not Started']}</td>
+      <td>{statuses['In Progress']}</td>
+      <td>{statuses.Closed}</td>
+      <td>{statuses.Suspended}</td>
+      <td>{total}</td>
+    </tr>
+  );
+}
+
+TopicRow.propTypes = {
+  topic: PropTypes.string.isRequired,
+  statuses: PropTypes.shape({
+    'Not Started': PropTypes.number,
+    'In Progress': PropTypes.number,
+    Closed: PropTypes.number,
+    Suspended: PropTypes.number,
+  }).isRequired,
+  total: PropTypes.number.isRequired,
+};
+
+export function TopicsTableWidget({ data, loading }) {
+  const [sortConfig, setSortConfig] = useSessionSort({
+    sortBy: 'topic',
+    direction: 'desc',
+    activePage: 1,
+  }, 'topicsTableWidget');
+  const [sorted, setSorted] = useState(data);
+
+  const requestSort = (sortBy) => {
+    let direction = 'desc';
+    if (sortConfig.sortBy === sortBy && sortConfig.direction === 'desc') {
+      direction = 'asc';
+    }
+    setSortConfig({ sortBy, direction, activePage: 1 });
+  };
+
+  useEffect(() => {
+    if (!data || !data.length) return;
+    const sortedData = data.sort((a, b) => {
+      const { sortBy, direction } = sortConfig;
+      let prop = sortBy;
+      let objA = a;
+      let objB = b;
+
+      if (sortBy.includes('statuses.')) {
+        [, prop] = sortBy.split('.');
+        objA = a.statuses;
+        objB = b.statuses;
+      }
+
+      if (objA[prop].localeCompare) {
+        return objA[prop].localeCompare(objB[prop]) * (direction === 'asc' ? 1 : -1);
+      }
+
+      if (objA[prop] > objB[prop]) {
+        return direction === 'asc' ? 1 : -1;
+      }
+
+      if (objA[prop] < objB[prop]) {
+        return direction === 'asc' ? -1 : 1;
+      }
+
+      return 0;
+    });
+    setSorted([...sortedData]);
+  }, [sortConfig, data]);
+
+  const getClassNamesFor = (name) => (sortConfig.sortBy === name ? sortConfig.direction : '');
+
+  const renderColumnHeader = (displayName, name) => {
+    const sortClassName = getClassNamesFor(name);
+    let fullAriaSort;
+    switch (sortClassName) {
+      case 'asc':
+        fullAriaSort = 'ascending';
+        break;
+      case 'desc':
+        fullAriaSort = 'descending';
+        break;
+      default:
+        fullAriaSort = 'none';
+        break;
+    }
+
+    return (
+      <th scope="col" aria-sort={fullAriaSort}>
+        {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+        <a
+          role="button"
+          tabIndex={0}
+          onClick={() => {
+            requestSort(name);
+          }}
+          onKeyDown={() => requestSort(name)}
+          className={`sortable ${sortClassName}`}
+          aria-label={`${displayName}. Activate to sort ${sortClassName === 'asc' ? 'descending' : 'ascending'
+          }`}
+        >
+          {displayName}
+        </a>
+      </th>
+    );
+  };
+
+  if (loading) return (<div>Loading...</div>);
```

Comment:
Sorry think this is the offending loader.

Reply:
fixed: [f9b97c6](https://github.com/HHS/Head-Start-TTADP/pull/1373/commits/f9b97c60ec04d1ace52e5643f4df7e4ff46fe3f7)